### PR TITLE
Remove all remaining references to AppVeyor

### PR DESCRIPTION
**Pull request by y-guyon**
It was replaced with .github/workflows/ci-windows-artifacts.yml.
Also remove the broken Travis Build Status link.

### CODE REVIEW

**Review by maryla-uc — APPROVED** [review]
yay

**Comment by y-guyon** [comment]
I do not understand why `continuous-integration/appveyor/pr` is still part of the presubmit tests.

File: README.md [hunk 1]
```diff
@@ -144,12 +144,10 @@ cmake -S . -B build -DBUILD_SHARED_LIBS=OFF -DAVIF_CODEC_AOM=LOCAL -DAVIF_LIBYUV
 cmake --build build --parallel
 ```
 
-## Prebuilt Library (Windows)
+## Prebuilt Static Binaries (Windows)
```
Comment (wantehchang) [inline README.md:147]: Nit: I suggest removing the word "Static".

It's not clear what "Static" means here. I think it means "statically-linked", i.e., the executables are linked with the libavif static library.

Comment (y-guyon) [reply README.md:147]: Done in https://github.com/AOMediaCodec/libavif/pull/2462.

Comment (y-guyon) [reply README.md:147]: Also done in https://github.com/AOMediaCodec/libavif/pull/2463.

**Pull request merged**